### DISCUSSION TITLE
When I call approval() QRCodeModal is open but when I scan QRcode nothing happen

### DISCUSSION THREAD

**Discussion by OstapDutkevych** [WalletConnect/walletconnect-monorepo#1655] [Developer Support]
Describe the issue
When I call approval() QRCodeModal is open but when I scan QRcode nothing happen

SDK Version (if relevant)
"@walletconnect/sign-client": "^2.1.1",

To Reproduce
To reproduce the behavior follow the instructions on https://docs.walletconnect.com/2.0/javascript/sign/installation

Create a new application
Install the packages above
Initialize the startup code from the installation page

`const signClient = await SignClient.init({
projectId: "65540ae88ea4b0258d3f53b4c7f0aacb",
});

try {
const {uri, approval} = await signClient.connect({
requiredNamespaces: {
eip155: {
methods: [
"eth_sendTransaction",
"eth_signTransaction",
"eth_sign",
"personal_sign",
"eth_signTypedData",
],
chains: ["eip155:1"],
events: ["chainChanged", "accountsChanged"],
},
},
});
if (uri) {
QRCodeModal.open(uri, () => {
console.log("EVENT", "QR Code Modal closed");
});
}

  // Await session approval from the wallet.
  const session = await approval();
  // Handle the returned session (e.g. update UI to "connected" state).
  // await onSessionConnected(session);
  console.log(session)
} catch (e) {
  console.error(e);
} finally {
  // Close the QRCode modal in case it was open.
  QRCodeModal.close();
}
`

Expected behavior
Should create session after I scan QRcode

Desktop (please complete the following information):

OS: MacOS
Browser Chrome
Additional context
maybe i'm doing something wrong
Could you please tell me what the problem is?
Please let me know how to best lend a hand

**Comment by chadyj** [comment]
What wallet are you using?
The wallet needs to be compatible with v2.
You can try with the sample wallet here https://react-wallet.walletconnect.com/

**Reply by chinmayHefty** [reply]
I used the same code mentioned above it shows the same problem. don't know what to do. please help

**Comment by adrianotadao** [comment]
I'm facing the same problem here. 

@chinmayHefty have you considered another option? Actually, is there another option? I have the feeling that nothing is ready to be used, which is quite frustrating.
